solr/core/src/java/org/apache/solr/search/MaxScoreCollector.java

```diff
@@ -39,7 +39,7 @@ public float getMaxScore() {
   public ScoreMode scoreMode() {
     // Should be TOP_SCORES but this would wrap the scorer unnecessarily since
     // this collector is only used in a MultiCollector.
-    return ScoreMode.COMPLETE;
+    return ScoreMode.TOP_SCORES;
```

Comment:
@jpountz, What did you mean with this comment? MultiCollector will set the `scoreMode` to `COMPLETE` in the case of the main collector being something other than `TOP_SCORES`, right?

Reply:
Yeah I think that at some point we tried not to wrap the scorer when all score modes were `COMPLETE` but apparently now we do, so this comment is stale. https://github.com/apache/lucene-solr/blob/master/lucene/core/src/java/org/apache/lucene/search/MultiCollector.java#L158